Makefile

```diff
@@ -1,82 +1,93 @@
 
 main.o : main.cpp
-	g++ ${CFLAGS} -c main.cpp
+	${CC} ${CFLAGS} -c $<
 
 utilities.o : devices/utilities.c
-	g++ ${CFLAGS} -c devices/utilities.c
+	${CC} ${CFLAGS} -c $<
 
 registers.o : devices/cpu/registers.c
-	g++ ${CFLAGS} -c devices/cpu/registers.c
+	${CC} ${CFLAGS} -c $<
 
 memspace.o : devices/memory/memspace.c
-	g++ ${CFLAGS} -c devices/memory/memspace.c
+	${CC} ${CFLAGS} -c $<
 
 debugger.o : debugger/debugger.c
-	g++ ${CFLAGS} -c debugger/debugger.c
+	${CC} ${CFLAGS} -c $<
 
 disassembler.o : debugger/disassembler.c
-	g++ ${CFLAGS} -c debugger/disassembler.c
+	${CC} ${CFLAGS} -c $<
 
 register_display.o : debugger/register_display.c
-	g++ ${CFLAGS} -c debugger/register_display.c
+	${CC} ${CFLAGS} -c $<
 
 decoder.o : devices/cpu/decoder.c
-	g++ ${CFLAGS} -c devices/cpu/decoder.c
+	${CC} ${CFLAGS} -c $<
 
 flag_handler.o : devices/cpu/flag_handler.c
-	g++ ${CFLAGS} -c devices/cpu/flag_handler.c
+	${CC} ${CFLAGS} -c $<
 
 formatI.o : devices/cpu/formatI.c
-	g++ ${CFLAGS} -c devices/cpu/formatI.c
+	${CC} ${CFLAGS} -c $<
 
 formatII.o : devices/cpu/formatII.c
-	g++ ${CFLAGS} -c devices/cpu/formatII.c
+	${CC} ${CFLAGS} -c $<
 
 formatIII.o : devices/cpu/formatIII.c
-	g++ ${CFLAGS} -c devices/cpu/formatIII.c
+	${CC} ${CFLAGS} -c $<
 
 bcm.o : devices/peripherals/bcm.c
-	g++ ${CFLAGS} -c devices/peripherals/bcm.c
+	${CC} ${CFLAGS} -c $<
 
 timer_a.o : devices/peripherals/timer_a.c
-	g++ ${CFLAGS} -c devices/peripherals/timer_a.c
+	${CC} ${CFLAGS} -c $<
 
 usci.o : devices/peripherals/usci.c
-	g++ ${CFLAGS} -c devices/peripherals/usci.c
+	${CC} ${CFLAGS} -c $<
 
 port1.o : devices/peripherals/port1.c
-	g++ ${CFLAGS} -c devices/peripherals/port1.c
+	${CC} ${CFLAGS} -c $<
+
+io.o: debugger/io.c
+	${CC} ${CFLAGS} -c $<
 
 emu_server.o : debugger/websockets/emu_server.cpp
-	g++ ${CFLAGS} -c debugger/websockets/emu_server.cpp
+	${CC} ${CFLAGS} -c $<
 
 packet_queue.o : debugger/websockets/packet_queue.c
-	g++ ${CFLAGS} -c debugger/websockets/packet_queue.c
+	${CC} ${CFLAGS} -c debugger/websockets/packet_queue.c
 
 # Server Program
 
-SERVER : server.o
-	g++ ${CFLAGS} -o server server.o -lrt -lpthread -lwebsockets -lssl -lcrypto;
+${SERVER} : server.o
+	${CC} ${CFLAGS} -o $@ $^ -lrt -lpthread -lwebsockets -lssl -lcrypto;
 
 server.o : debugger/server/server.c
-	g++ ${CFLAGS} -c debugger/server/server.c
+	${CC} ${CFLAGS} -c $<
 
 
 clean :
-	rm server.o main.o utilities.o emu_server.o registers.o \
-	memspace.o debugger.o disassembler.o \
-	register_display.o decoder.o flag_handler.o formatI.o \
-	formatII.o formatIII.o \
-	usci.o port1.o packet_queue.o bcm.o timer_a.o \
-	*.bin *.tmp *.elf \
-	MSP430 server;
+	rm -f server.o main.o utilities.o emu_server.o registers.o \
+		memspace.o debugger.o disassembler.o \
+		register_display.o decoder.o flag_handler.o formatI.o \
+		formatII.o formatIII.o io.o \
+		usci.o port1.o packet_queue.o bcm.o timer_a.o \
+		${EMULATOR} ${SERVER}
+	make -C test clean
```

Comment:
somewhere I saw that calling `make` from `make` is bad, and `$(MAKE)` should be used... it supposed to be some "special treatment". No idea if its true or what are the consequences, just some vietnam flashback from my times in autotools related BS

Reply:
And yet, I have seen make calling make... and it works. But:
https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html
Ack